core/src/test/scala/besom/internal/ValueAssertions.scala

```diff
@@ -0,0 +1,30 @@
+package besom.internal
+
+import com.google.protobuf.struct.*, Value.Kind
+import Constants.*
+import ProtobufUtil.*
+
+trait ValueAssertions:
+  self: munit.Assertions =>
+
+  def assertEqualsMaybeValue(actual: Option[Value], expected: Option[Value]): Unit =
+    (actual, expected) match
+      case (None, None) => ()
+      case (None, Some(Value(Kind.NullValue(_), _))) => ()
+      case (Some(Value(Kind.NullValue(_), _)), None) => ()
+      case (Some(a), Some(b)) => assertEqualsValue(a, b)
+      case _ => throw Exception(s"Values not equal: $actual != $expected")
```

Comment:
These helpers don't really seem to be helpful IMO. Not only we lose some information about how exactly we (de)serialize things but also in case of an error it's not clear where something was wrong. Relying on `munit`'s `assertEquals` seems a better idea to me as it shows diffs in actual structures of data